### PR TITLE
Breaking - Update button icon functionality to use calcite-icon and calcite UI icon names

### PR DESCRIPTION
**Pull request by macandcheese**
Fixes: https://github.com/Esri/calcite-components/issues/297

Breaking Change:

- Replace acceptance of svg path data with calcite ui icon name string
- Scale icons with scale of button
- Update changelog, storybook, dev demo examples

### CODE REVIEW

File: src/components/calcite-button/calcite-button.stories.js [hunk 1]
```diff
@@ -2,9 +2,21 @@ import { storiesOf } from "@storybook/html";
 import { withKnobs, text, boolean, select } from "@storybook/addon-knobs";
 import { images24 } from "@esri/calcite-ui-icons/js/images24";
 import { darkBackground, parseReadme } from "../../../.storybook/helpers";
+import * as icons from "../../../node_modules/@esri/calcite-ui-icons";
```
Comment (paulcpederson) [inline src/components/calcite-button/calcite-button.stories.js]: this is going to be massive, but for docs maybe that's ok

File: src/components/calcite-button/calcite-button.tsx [hunk 2]
```diff
@@ -138,15 +138,22 @@ export class CalciteButton {
       </div>
     );
 
-    const icon = (
-      <svg
+    const iconScale =
```
Comment (paulcpederson) [inline src/components/calcite-button/calcite-button.tsx]: I sometimes find it easier to just make a lookup for things like this as the ternaries can get pretty complicated:

```
const iconScales = {
  xs: "s",
  s: "s",
  m: "s",
  l: "m"
}

iconScales[this.scale] || "l"
```

**Comment by paulcpederson** [comment]
nice! this is a great improvement

File: src/components/calcite-button/calcite-button.tsx [hunk 3]
```diff
@@ -73,7 +73,7 @@ export class CalciteButton {
   /** optionally pass a href - used to determine if the component should render as a button or an anchor */
   @Prop({ reflect: true }) href?: string;
 
-  /** optionally pass icon path data - pass only raw path data from calcite ui helper  */
+  /** optionally pass an icon to display - accepts calcite ui icon names  */
```
Comment (jcfranco) [inline src/components/calcite-button/calcite-button.tsx]: Nitpick: UI should be uppercased, no?

File: src/components/calcite-button/calcite-button.tsx [hunk 4]
```diff
@@ -138,15 +138,22 @@ export class CalciteButton {
...
         class="calcite-button--icon"
-        xmlns="http://www.w3.org/2000/svg"
-        preserveAspectRatio="xMidYMid meet"
-        viewBox="0 0 24 24"
-      >
-        <path d={this.icon} />
-      </svg>
+        icon={this.icon}
+        scale={iconScale}
+      ></calcite-icon>
```
Comment (jcfranco) [inline src/components/calcite-button/calcite-button.tsx]: Nitpick: This can be a self-closing tag.

File: src/demos/calcite-button.html [hunk 5]
```diff
@@ -352,17 +351,24 @@ <h3>Scales</h3>
   <calcite-button title="s scale button" scale='s'>s scale button</calcite-button>
   <calcite-button title="xs scale button" scale='xs'>xs scale button</calcite-button>
 
+  <h3>Scales with icons</h3>
+  <calcite-button title="xl scale button" icon="check-circle" scale='xl'>xl scale button</calcite-button>
+  <calcite-button title="l scale button" icon="check-circle" scale='l'>l scale button</calcite-button>
+  <calcite-button title="m / default scale button" icon="check-circle">m scale button</calcite-button>
```
Comment (jcfranco) [inline src/demos/calcite-button.html]: Nitpick: `m (default) ...`? Related, does the text in this button also need to mention it's the default?

File: src/components/calcite-button/calcite-button.stories.js [hunk 6]
```diff
@@ -2,9 +2,21 @@ import { storiesOf } from "@storybook/html";
...
 import readme from "./readme.md";
 const notes = parseReadme(readme);
 
+// we can get all unique icon names from all size 16 non-filled icons.
+const iconNames = Object.keys(icons)
```
Comment (jcfranco) [inline src/components/calcite-button/calcite-button.stories.js]: I could create a utility for icons since this is now in two stories (`icon` and here).

Comment (macandcheese) [reply src/components/calcite-button/calcite-button.stories.js]: 👍 That'd be useful for upcoming components that accept icons as well.

Comment (driskull) [reply src/components/calcite-button/calcite-button.stories.js]: agreed!!

Comment (jcfranco) [reply src/components/calcite-button/calcite-button.stories.js]: Created issue: https://github.com/Esri/calcite-components/issues/336https://github.com/Esri/calcite-components/issues/336

**Pull request merged**